### PR TITLE
[LW-6640] build repo with yarn v3

### PR DESCRIPTION
**Pull request by tgunnoe**
# Checklist

- [x] JIRA - https://input-output.atlassian.net/browse/LW-6640
- [x] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution
Minimal upgrade path to yarn v3, keeping node_modules as is and not switching to Plug n Play modules present since yarn berry.

A lot of the scripts needed to be adjusted for the new cli.  I was also able to remove a few dependencies relating to workspaces and executing commands, because yarn v3 has replacements for these. I also moved the `prepare` script to `postinstall`, as yarn doesn't run prepare after install now.

## Screenshots

Attach screenshots here if implementation involves some UI changes

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ✅ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/linux/chrome/1482/5869676952/index.html) for [f75d31d7](https://github.com/input-output-hk/lace/pull/159/commits/f75d31d734d8e80c59078da9e4d535d2027245ed)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 35     | 0      | 0       | 0     | 35    | ✅     |

### CODE REVIEW

**Review by mchappell — APPROVED** [review]

File: .github/shared/build/action.yml [hunk 1]
```diff
@@ -21,9 +21,9 @@ runs:
         key: ${{ runner.os }}-${{ hashFiles('yarn.lock') }}
     - name: Install dependencies
       shell: bash
-      run: yarn install --frozen-lockfile --non-interactive --logevel=error
+      run: yarn install --immutable
```
Comment (rhyslbw) [inline .github/shared/build/action.yml:24]:
```suggestion
      run: yarn install --immutable --inline-builds
```

https://github.com/input-output-hk/cardano-js-sdk/blob/ab2265aa118b7f771055b5e3eab5621eea4f3476/.github/workflows/continuous-integration-unit-tests.yaml#L31

Maybe it's helpful to use this option too https://yarnpkg.com/cli/install#options

File: apps/browser-extension-wallet/package.json [hunk 2]
```diff
@@ -25,18 +25,18 @@
...
   ],
   "scripts": {
-    "build": "NODE_OPTIONS='--openssl-legacy-provider' rm -rf dist & webpack --config webpack.sw.prod.js --progress & webpack --config webpack.app.prod.js --progress",
-    "build:dev": "NODE_OPTIONS='--openssl-legacy-provider' rm -rf dist & webpack --config webpack.sw.dev.js --progress & webpack --config webpack.app.dev.js --progress",
-    "cleanup": "shx rm -rf dist node_modules",
-    "dev": "NODE_OPTIONS='--openssl-legacy-provider' rm -rf dist & webpack --config webpack.sw.dev.js --progress --watch & webpack serve --config webpack.app.dev.js --env RUN_DEV_SERVER=true",
-    "lint": "cd ../.. && yarn extension:lint",
+    "build": "NODE_OPTIONS='--openssl-legacy-provider' rm -rf dist & run -T webpack --config webpack.sw.prod.js --progress & run -T webpack --config webpack.app.prod.js --progress",
+    "build:dev": "NODE_OPTIONS='--openssl-legacy-provider' rm -rf dist & run -T webpack --config webpack.sw.dev.js --progress & run -T webpack --config webpack.app.dev.js --progress",
+    "cleanup": "yarn exec rm -rf dist node_modules",
```
Comment (rhyslbw) [inline apps/browser-extension-wallet/package.json:30]: This is nice. We can make the same change in the `cardano-js-sdk` and drop `shx` as a dependency

**Review by rhyslbw — COMMENTED** [review]
LGTM

File: packages/e2e-tests/README.md [hunk 3]
```diff
@@ -27,7 +27,7 @@ UI-mapped gherkin tests for the Lace browser extension
 
 - Build the extension (from the project root)
 
-  - `yarn browser build`
+  - `yarn build`
 
 - Run tests (from the `packages/e2e-tests` directory)
```
Comment (ljagiela) [inline packages/e2e-tests/README.md:32]: I think we need to remove all occurrences of `./node_modules/.bin/` in this file

**Review by ljagiela — CHANGES_REQUESTED** [review]

File: .github/shared/build/action.yml (hunk 1 above)
Comment (tgunnoe) [reply .github/shared/build/action.yml]: yes, thank you.

File: packages/e2e-tests/README.md (hunk 3 above)
Comment (tgunnoe) [reply packages/e2e-tests/README.md:32]: thank you!

**Review by tshaynik — APPROVED** [review]
LGTM.

**Review by ljagiela — APPROVED** [review]

**Review by miorsufianiohk — APPROVED** [review]

**Pull request merged**